src/Editor/SourceView.tsx

```diff
@@ -4,6 +4,37 @@ import { getCode } from "../JavascriptVM/vmSlice";
 import SyntaxHighlighter from "react-syntax-highlighter";
 
 import "./SourceView.css";
+import { BLOCKLY_HIGHLIGHT_PREFIX } from "../BlocklyInterface/BlocklyInstance";
+
+/**
+ * Removes any occurrences of the BLOCKLY_HIGHLIGHT_PREFIX in the
+ * provided code.
+ *
+ * @param blocklyCode Code that might contain BLOCKLY_HIGHLIGHT_PREFIX
+ * statements.
+ *
+ * @returns A string that is identical to the provided 'blocklyCode',
+ * but without BLOCKLY_HIGHLIGHT_PREFIX statements.
+ */
+function removeHighlightBlockCalls(blocklyCode: string | null): string {
+  if (!blocklyCode) return "";
+
+  const blocklyCodeLines: Array<string> = blocklyCode.split("\n");
+  let output: string = "";
+  for (let i = 0; i < blocklyCodeLines.length; ++i) {
+    if (blocklyCodeLines[i].includes(BLOCKLY_HIGHLIGHT_PREFIX)) {
```

Comment:
This is a simplistic implementation that relies on the `highlightBlock` statements to be on their own line.

I think this comes from two things:
* We append a newline as part of our `STATEMENT_PREFIX`, [here](https://github.com/FRUK-Simulator/Simulator/blob/master/src/BlocklyInterface/BlocklyInstance.ts#L38)
* The blockly generator separates each section via newlines, [here](https://github.com/google/blockly/blob/93b5be40b6229cf164e83b63fe5a6cc4d44a9b5f/core/generator.js#L80-L122)

If we want to want to further future-proof the filtering code we could try to just match  `highlightBlock('*');` or `highlightBlock("*");` via a regex.  It seems the characters that can occur in the unique ID are documented, [here](https://github.com/google/blockly/blob/master/core/utils.js#L423)

Let me know what you'd prefer.

Reply:
I think we can happily rely on them being on their own lines :). i think a regex which matches any line where the highlightBlock string appears can be removed. regex: `^\w*highlightBlock.*$` or something like that will work.